plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/m2e/core/ui/internal/dialogs/UpdateMavenProjectsDialog.java

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Red Hat, Inc and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.reddeer.eclipse.m2e.core.ui.internal.dialogs;
+
+import org.eclipse.reddeer.common.wait.TimePeriod;
+import org.eclipse.reddeer.common.wait.WaitWhile;
+import org.eclipse.reddeer.jface.dialogs.TitleAreaDialog;
+import org.eclipse.reddeer.swt.condition.ShellIsAvailable;
+import org.eclipse.reddeer.swt.impl.button.CancelButton;
+import org.eclipse.reddeer.swt.impl.button.CheckBox;
+import org.eclipse.reddeer.swt.impl.button.OkButton;
+import org.eclipse.reddeer.workbench.core.condition.JobIsRunning;
+
+/**
+ * Represends "Update Maven Project" dialog. Can operate OK and Cancel buttons and "Clean projects" checkbox.
+ * 
+ * @author lvalach
+ *
+ */
+public class UpdateMavenProjectsDialog extends TitleAreaDialog {
+
+	public static final String SHELL_TITLE = "Update Maven Project";
+	public static final String CLEAN_PROJECTS = "Clean projects";
+
+	public UpdateMavenProjectsDialog() {
+		super(SHELL_TITLE);
+	}
+
+	/**
+	 * Click "OK" button.
+	 */
+	public void ok() {
```

Comment:
can you please also add method ok(TimePeriod waitForJobs). TimePeriod.LONG may not be enough for all cases